File: src/main/webapp/Task.js

class Task {
    // taskId;
    // subject;
    // author;
    // performers;
    // period;
    // control;
    // execution;
    // descr;

    constructor(taskId, subject, author, performers, period, control, execution, descr) {
        this._taskId = taskId;
        this._subject = subject;
        this._author = author;
        this._performers = performers;
        this._period = period;
        this._control = control;
        this._execution = execution;
        this._descr = descr;
    }

    update () {
        return `<form id="taskForm">
                id <input name="taskId" type="text" value=${this._taskId}><br>
                Предмет поручения <input name="subject" type="text" value=${this._subject}><br>
                Автор поручения <input name="author" type="text" value=${this._author.fullName()}><br>
                Срок исполнения <input name="period" type="date" value=${this._period}><br>
                Признак контрольности <input name="control" type="checkbox" value=${this._control}><br>
                Признак исполнения <input name="execution" type="checkbox" value=${this._execution}><br>
                Текст поручения <input name="descr" type="text" value=${this._descr}><br>
                </form>`
    }

    render () {
        return `<h3><b>Карточка сотрудника</b></h3><br>
                Предмет поручения: ${this._subject}<br>
                Автор поручения: ${this._author.fullName()}<br>
                Срок исполнения: ${this._period}<br>
                Признак контрольности: ${this._control}<br>
                Признак исполнения: ${this._execution}<br>
                Текст поручения: ${this._descr}<br>

                `
    }

    addForm() {
        return `<form id="addTaskForm">
                Предмет поручения <input name="subject" type="text"><br>
                Срок исполнения <input name="period" type="date"><br>
                Признак контрольности <input name="control" type="checkbox"><br>
                Признак исполнения <input name="execution" type="checkbox"><br>
                Текст поручения <input name="descr" type="text"><br>
                </form>`
    }

    get taskId() {
        return this._taskId;
    }

    set taskId(value) {
        this._taskId = value;
    }

    get subject() {
        return this._subject;
    }

    set subject(value) {
        this._subject = value;
    }

    get author() {
        return this._author;
    }

    set author(value) {
        this._author = value;
    }

    get performers() {
        return this._performers;
    }

    set performers(value) {
        this._performers = value;
    }

    get period() {
        return this._period;
    }

    set period(value) {
        this._period = value;
    }

    get control() {
        return this._control;
    }

    set control(value) {
        this._control = value;
    }

    get execution() {
        return this._execution;
    }

    set execution(value) {
        this._execution = value;
    }

    get descr() {
        return this._descr;
    }

    set descr(value) {
        this._descr = value;
    }
}

File: src/main/java/ru/dinis/docs/beans/Task.java
package ru.dinis.docs.beans;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.xml.bind.annotation.XmlRootElement;
import java.io.Serializable;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;
import java.util.TreeSet;

/**
 * Create by dinis of 21.04.18.
 */
@Entity
@Table(name = "tasks")
@XmlRootElement
public class Task implements Serializable, Comparable<Task> {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "task_id")
    private int taskId;

    @Column(name = "subject")
    private String subject;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "empl_id")
    @JsonBackReference
    private Employee author;

    @ManyToMany
    @JoinTable(name = "empl_tasks",
            joinColumns = @JoinColumn(name = "task_id"),
            inverseJoinColumns = @JoinColumn(name = "empl_id"))
    @JsonIgnore
    private Set<Employee> performers = new TreeSet<>();

    @Column(name = "period")
    @Temporal(TemporalType.DATE)
    private Date period;

    @Column(name = "control")
    private Boolean control;

    @Column(name = "execution")
    private Boolean execution;

    @Column(name = "descr")
    private String descr;

    public Task() {
    }

    @Override
    public int compareTo(Task o) {
        return this.taskId > o.taskId ? 1 : -1;
    }

    public int getTaskId() {
        return taskId;
    }

    public void setTaskId(int taskId) {
        this.taskId = taskId;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public Employee getAuthor() {
        return author;
    }

    public void setAuthor(Employee author) {
        this.author = author;
    }

    public Set<Employee> getPerformers() {
        return performers;
    }

    public void setPerformers(Set<Employee> performers) {
        this.performers = performers;
    }

    public Date getPeriod() {
        return period;
    }

    public void setPeriod(Date period) {
        this.period = period;
    }

    public Boolean getControl() {
        return control;
    }

    public void setControl(Boolean control) {
        this.control = control;
    }

    public Boolean getExecution() {
        return execution;
    }

    public void setExecution(Boolean execution) {
        this.execution = execution;
    }

    public String getDescr() {
        return descr;
    }

    public void setDescr(String descr) {
        this.descr = descr;
    }
}
